### PR TITLE
chore(build): restructure evergreen config

### PR DESCRIPTION
**Pull request by rose-m**
Extracts the `compile_artifact` and `package_and_upload_artifact` into separate shell scripts. Note that it uses the `env:` section of `subprocess.exec` to pass on the `NODE_JS_VERSION` variable expansion.

### CODE REVIEW

File: .evergreen/package-and-upload-artifact.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (rose-m) [reply .evergreen/package-and-upload-artifact.sh:1]: Ah right 🙈 Just used to adding it :D :D

[the remark this answers is not in the document]
Comment (addaleax) [reply .evergreen/package-and-upload-artifact.sh:1]: To be clear, I think keeping it and changing the file modes is what makes sense here :)

[the remark this answers is not in the document]
Comment (rose-m) [reply .evergreen/package-and-upload-artifact.sh:1]: I'll let evergreen do it's thing - and afterwards I'll `+x` them just before merging 👍

**Comment by addaleax** [comment]
@rose-m Thank you! :)

**Pull request merged**